kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ElementAtEvaluator.java

```diff
@@ -73,7 +74,7 @@ static ColumnVector eval(ColumnVector map, ColumnVector lookupKey) {
             // The general pattern is call `isNullAt(rowId)` followed by `getString`.
             // So the cache of one value is enough.
             private int lastLookupRowId = -1;
-            private Object lastLookupValue = null;
+            private String lastLookupValue = null;
```

Comment:
what does `null` represent? that there was no last lookup value? this is ambiguous, it should be `Optional`

Reply:
Can't represent a `null` value if we make it `Optional<String>`

Reply:
I think `lastLookupRowId` reflects whether there was a last lookup value.